### PR TITLE
Generic block3

### PR DESCRIPTION
**Pull request by diptighosalkar**
Made generic_block3 working, added set and get method. 
Added needed validation for inputs (As per my understanding as in SCI file is not available)

### CODE REVIEW

File: webapp/data_structures/generic_block3.js [hunk 1]
```diff
@@ -4,25 +4,352 @@ function generic_block3() {
...
+        ci1 = MatrixInverse(ci1)
+        co1 = MatrixInverse(co1)
+        if (ci1.length == 0 || ci1[0].length == 0){
+            ci1 = [];
+        }
+        if (co1.length == 0 || co1[0].length == 0 ){
+            co1 = [];
+        }
+        if (ci1.length != 0 || co1.length != 0) {
+            if (math.max(ci1.concat(co1)) > 1) {
```
Comment (sunilshetye) [inline webapp/data_structures/generic_block3.js]: We need to discuss this test.

File: webapp/data_structures/generic_block3.js [hunk 2]
```diff
@@ -4,25 +4,352 @@ function generic_block3() {
...
+                alert("vector event links not supported");
+                throw "incorrect";
+            }
+        }
+        this.ci = ci1
```
Comment (sunilshetye) [inline webapp/data_structures/generic_block3.js]: add semicolon

assign after all validation checks

File: webapp/data_structures/generic_block3.js [hunk 3]
```diff
@@ -4,25 +4,352 @@ function generic_block3() {
...
+
+        auto01 = MatrixInverse(auto01)
+        if (auto01.length == 0 || auto01[0].length == 0){
+            auto01 = [];
+        }
+        this.auto0 = auto01
+        this.depu = depu1
+        this.dept = dept1
+        this.depu = this.depu.trim();
+        this.depu_boolean = false;
```
Comment (sunilshetye) [inline webapp/data_structures/generic_block3.js]: is this depu_boolean used anywhere?

File: webapp/data_structures/generic_block3.js (hunk 1 above)
Comment (sunilshetye) [reply webapp/data_structures/generic_block3.js]: test is correct.

File: webapp/data_structures/generic_block3.js [hunk 4]
```diff
@@ -4,25 +4,352 @@ function generic_block3() {
         var model = scicos_model();
         this.function_name = "sinblk";
         this.funtyp = 4;
+        this.in = [[1],[1]];
+        this.it = [1]
```
[the remark this answers is not in the document]
Comment (sunilshetye) [reply webapp/data_structures/generic_block3.js]:
```suggestion
        this.it = [1];
```

semicolon required here

File: webapp/data_structures/generic_block3.js [hunk 5]
```diff
@@ -4,25 +4,363 @@ function generic_block3() {
...
+
+        if (regex_colon.test(oz1)){
+            alert("Answer given for Initial object state is incorrect:\n It has to be type of list eg. list()");
+            throw "incorrect";
+        }
+        if (oz1.substring(0,5) != "list(" && oz1.charAt(oz1.length-1) != ")"){
+            alert("Answer given for Initial object state is incorrect");
+            throw "incorrect";
+        }
+        this.oz = oz1;
```
Comment (sunilshetye) [inline webapp/data_structures/generic_block3.js]: this should also be moved after the validation checks.

File: webapp/data_structures/generic_block3.js [hunk 6]
```diff
@@ -4,25 +4,363 @@ function generic_block3() {
...
+
+        if (regex_colon.test(opar1)){
+            alert("Answer given for Object parameters list is incorrect:\n It has to be type of list eg. list()");
+            throw "incorrect";
+        }
+        if (opar1.substring(0,5) != "list(" && opar1.charAt(opar1.length-1) != ")"){
+            alert("Answer given for Object parameters list is incorrect");
+            throw "incorrect";
+        }
+        this.opar = opar1;
```
Comment (sunilshetye) [inline webapp/data_structures/generic_block3.js]: this should also be moved after the validation checks.

File: webapp/data_structures/generic_block3.js (hunk 5 above)
Comment (diptighosalkar) [reply webapp/data_structures/generic_block3.js]: these is purposely kept here as this.oz contain value 'list(1,2,1)' which will be used in get method but when it is assign to odstate it has to be scilab list type with only passing values 1,2,1. After assign value to this.oz i do all modification to oz1 and pass that to model object's odstate

File: webapp/data_structures/generic_block3.js (hunk 6 above)
Comment (diptighosalkar) [reply webapp/data_structures/generic_block3.js]: same as odstate opar works

**Pull request merged**